Sources/MapboxMaps/Foundation/BaseMapView.swift

```diff
@@ -333,6 +338,15 @@ open class BaseMapView: UIView, MapClient, MBMMetalViewProvider, CameraViewDeleg
         if needsDisplayRefresh {
             needsDisplayRefresh = false
             self.cameraView.update()
+
+            /// This executes the series of scheduled animation completion blocks and also removes them from the list
+            while !pendingAnimatorCompletionBlocks.isEmpty {
+                let pendingCompletion = pendingAnimatorCompletionBlocks.removeFirst()
```

Comment:
why are you only removing the first? We should be executing and and all pending completions here.

Reply:
removeFirst will remove the first out of the array so it utilizes a queue based system. This way I am maintaining order while also removing from the array when I am executing the function.

The while loop will continue to execute all blocks in the array until it is empty

Reply:
ah i missed the `while`.